### ISSUE TITLE
short method references

### ISSUE THREAD

**Issue by olivernybroe** [php-plus/engine#2]
So in PHP we can make a method reference by using this syntax right now
```
Route::get('users', [UserController::class, 'index']);
```

I propose that we add a short method reference syntax like so 
```
Route::get('users', UserController::index);
```

**Comment by stancl** [comment]
Should this work only with static access? It could be problematic with access inside the class because you can have a property and a method with the same name. `$this->foo()` and `($this->foo)()` are different things.

And with static methods, this is already possible. Strings that reference static methods are callable:
```php
>>> class Bar { public static function xyz() { return 'foo'; } }
>>> 'Bar::xyz'();
=> "foo"
```

**Comment by olivernybroe** [comment]
@stancl This should work with non static method also. It should not be used for properties, only methods.
As in my example, it would basically just be a wrapper for `[Bar::class, 'xyz']`, as that syntax also supports static methods.
```
>>> class Bar { public static function xyz() { return 'foo'; } }
>>> [Bar::class, 'xyz']();
=> "foo"
```

The only "problem" I really see with it is that the same syntax is used for constants. However in that case, I think the constant should have priority.

**Comment by enumag** [comment]
> The only "problem" I really see with it is that the same syntax is used for constants. However in that case, I think the constant should have priority.

In real world constants are usually written in uppercase and methods in camelcase so it's not a real problem. In my opinion Plus should simply throw an error if a class has both a method and a constant with the same name.

**Comment by olivernybroe** [comment]
@enumag Yeah, I like the idea of just throwing an error. And the uppercase/camelcase scenario was also why I sad "problem" 👍
